plugins/single_plugins/idle.cpp

```diff
@@ -4,69 +4,297 @@ extern "C"
 }
 
 #include "singleton-plugin.hpp"
+#include "render-manager.hpp"
 #include "output.hpp"
 #include "core.hpp"
 #include "config.hpp"
 #include "output-layout.hpp"
+#include "../cube/cube-control-signal.hpp"
+
+#include <cmath>
+#include <animation.hpp>
+
+#define ZOOM_BASE 1.0
+
+enum screensaver_state
+{
+    SCREENSAVER_DISABLED,
+    SCREENSAVER_RUNNING,
+    SCREENSAVER_STOPPING
+};
 
 class wayfire_idle
 {
+    wf::effect_hook_t render_hook;
+    double rotation = 0.0;
+    wf_duration duration;
+    wf_transition rot_transition, zoom_transition;
     bool idle_enabled = true;
-    wlr_idle_timeout *timeout = NULL;
-    wf::wl_listener_wrapper on_idle, on_resume;
+    screensaver_state state = SCREENSAVER_DISABLED;
+    std::map<wf::output_t*, bool> hook_set;
```

Comment:
Can we have a better name for this? Maybe output_inhibited?

Reply:
It can be renamed but output_inhibited is a bad name because this is actually true when the output is not inhibited and the cube screensaver is running.

Reply:
I see. So what about screensaver_enabled?

Reply:
I went with screensaver_hook_set since that's what it means.